ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OWord2VecBase.scala

```diff
@@ -0,0 +1,50 @@
+package ai.h2o.sparkling.ml.features
+
+import ai.h2o.sparkling.H2OFrame
+import ai.h2o.sparkling.ml.models.H2OWord2VecMOJOModel
+import ai.h2o.sparkling.ml.params.H2OWord2VecExtraParams
+import hex.Model
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.functions.{col, explode, size, udf}
+
+import scala.reflect.ClassTag
+
+abstract class H2OWord2VecBase[P <: Model.Parameters: ClassTag]
+  extends H2OFeatureEstimator[P]
+  with H2OWord2VecExtraParams {
+
+  private val appendEmptyStringUdf: UserDefinedFunction = udf[Seq[String], Seq[String]](_ :+ "")
+
+  override def fit(dataset: Dataset[_]): H2OWord2VecMOJOModel = {
+    val inputCol: String = getInputCol()
+    val ds = dataset
+      .filter(col(inputCol).isNotNull)
+      .filter(size(col(inputCol)) =!= 0)
+      .withColumn(inputCol, appendEmptyStringUdf(col(inputCol)))
```

Comment:
the old impl did that and till this point I'm not 100% sure what's the reason behind that

Reply:
it seems that empty string is used as a delimiter of sentences. explode will crate a string column from an array column. See https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-munging/tokenize.html#tokenize